### ISSUE TITLE
3d model quadcopters not visible

### ISSUE THREAD

**Issue by knmcguire** [malintha/multi_uav_simulator#10]
Hi!

Awesome simulation! But I ran against small problem. It seems that the quadcopter models are not visible when I try it on my computer eventhough I do find the .dae files in the directory downloaded with the repo:

OS: Ubuntu 20.04
ROS noetic

![rviz_screenshot_2022_02_18-16_58_12](https://user-images.githubusercontent.com/11061260/154717448-e80afa5d-da6d-41ce-afa1-23dbebb93eff.png)

got some log files here but they are not indicating any errors.

[master.log](https://github.com/malintha/multi_uav_simulator/files/8098549/master.log)
[roslaunch-kimberly-ThinkPad-T460p-55689.log](https://github.com/malintha/multi_uav_simulator/files/8098550/roslaunch-kimberly-ThinkPad-T460p-55689.log)

**Comment by malintha** [comment]
Hey,

Thanks for trying it out and opening the issue! I can see the ros params for the drones being published in the master.log. They are supposed store the rendered urdf output and pass that into the corresponding robotmodel.
```
[rosmaster.master][INFO] 2022-02-18 16:58:01,210: +PARAM [/cf1] by /roslaunch
[rosmaster.master][INFO] 2022-02-18 16:58:01,211: +PARAM [/cf2] by /roslaunch
[rosmaster.master][INFO] 2022-02-18 16:58:01,211: +PARAM [/cf3] by /r
![Selection_002](https://user-images.githubusercontent.com/3253761/154789302-7e22a7d2-3f38-4024-9b81-07df60c228d1.png)
oslaunch
[rosmaster.master][INFO] 2022-02-18 16:58:01,211: +PARAM [/cf4] by /roslaunch
[rosmaster.master][INFO] 2022-02-18 16:58:01,211: +PARAM [/cf5] by /roslaunch
```
I suspect it could be that the RobotModels missing from Rviz when populating it with the `simulation.rviz` file (although strange that we can see all the other markers). There should be 5 RobotModel elements with the description as cf1 - cf5. Can you please check if they are there in the displays pane in Rviz?

**Comment by knmcguire** [comment]
It seems to be pretty similar to your display tab as well. Bit strange... 

Can also be related to my opengl version? Thats

```
$ glxinfo | grep "OpenGL version"
OpenGL version string: 4.6 (Compatibility Profile) Mesa 20.2.6

```

And the ros print out also had these lines:
```
[ INFO] [1645456079.199418091]: rviz version 1.14.10
[ INFO] [1645456079.199477225]: compiled against Qt version 5.12.8
[ INFO] [1645456079.199492299]: compiled against OGRE version 1.9.0 (Ghadamon)
[ INFO] [1645456079.210905179]: Forcing OpenGl version 0.
[ INFO] [1645456079.517119766]: Stereo is NOT SUPPORTED
[ INFO] [1645456079.517189228]: OpenGL device: Mesa Intel(R) HD Graphics 530 (SKL GT2)
[ INFO] [1645456079.517220293]: OpenGl version: 4,6 (GLSL 4,6) limited to GLSL 1.4 on Mesa system.
```

Wonder what the 'limited to GLSL 1.4' implies too.. perhaps this is a generic RVIZ bug?

[Screenshot from 2022-02-21 16-09-14](https://user-images.githubusercontent.com/11061260/154981869-4cd3612c-84f4-4482-809f-53d4fb46aaf1.png)

**Comment by malintha** [comment]
Apparently, it is somewhat of a new bug with Rviz and certain graphic drivers. I tested on a fresh ROS Noetic installation on an Ubuntu 20.04 Laptop, also with Intel graphics. The CL output was pretty similar to yours, but it displayed the robot models.
```
[ INFO] [1645334491.049737087]: rviz version 1.14.14
[ INFO] [1645334491.049779913]: compiled against Qt version 5.12.8
[ INFO] [1645334491.049786825]: compiled against OGRE version 1.9.0 (Ghadamon)
[ INFO] [1645334491.055008469]: Forcing OpenGl version 0.
[ INFO] [1645334491.191081338]: Stereo is NOT SUPPORTED
[ INFO] [1645334491.191120985]: OpenGL device: Mesa Intel(R) Xe Graphics (TGL GT2)
[ INFO] [1645334491.191140222]: OpenGl version: 4.6 (GLSL 4.6) limited to GLSL 1.4 on Mesa system.
```
With a bit of searching it seems like we can force RViz to use software graphics rendering when the dedicated graphics is not present by setting `export LIBGL_ALWAYS_SOFTWARE=1`. Apparently, it is a temporary solution, hope it will be fixed soon.
Found these instructions at https://github.com/ros-planning/moveit/issues/3006 and https://github.com/microsoft/wslg/issues/554.

**Comment by knmcguire** [comment]
Yes I remembered that from when I was trying out ros from WSL and xserver. But this is on a Ubuntu 20.04, and I tried to export the same variable and it has not solved it, so there is probably something else going on...

Anyway, I'll  try to investigate this a bit more. At least this issue can warn people if they are seeing the same thing
